src/PIL/GimpPaletteFile.py

```diff
@@ -49,7 +49,7 @@ def __init__(self, fp):
                 warnings.warn(
                     f"Palette file truncated at {self._max_file_size - len(s)} bytes"
                 )
-                break
+                break  # pragma: no cover
```

Comment:
Why is this "no cover"? I know your commit message says "testing promotes the warning to an exception", but I don't know what you mean by that. All our CIs passed.

Reply:
Hi - I also had to learn further on Warnings to write this code: 
The thing is, on ordinary runs, creating a warning just does its thing - with no further configurations it logs to stderr. 

However, the Warnings has a statefull config (which takes place in a `with` block in the test), which changes then to raise as exceptions instead of simply logging. I found it simpler to check for the warning by letting it raise and using `pytest.raies`. 

The problem is the coverage extension: on running tests, as the warning raises, line 52 is not executed: the pipelines warned me that this line was not covered in tests. the `# pragma: no cover`  is the explict way to tell the coverage plug-in it should not worry about this line.

Of course, I don't know at this point if it is preferrable to have a "not 100% covered" report, as that may be a misleading metric, to have the visual dirt these directives insert - feel free to remove it.

Reply:
The solution to this is to use [`pytest.warns`](https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html#warns), as Pillow does [elsewhere](https://github.com/python-pillow/Pillow/blob/e0c30d2fcd90ed9cfb1bdf2d27f6170bc9e762ad/Tests/test_file_jpeg.py#L860-L861). I've created https://github.com/jsbueno/Pillow/pull/3, which will cover the line.